### ISSUE TITLE
brew update broke octave

### ISSUE THREAD

**Issue by houtanb** [Homebrew/homebrew-science#4366]
### Indicate troubleshooting steps taken

- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [X] Checked that a similar issue hasn't already been filed?
- [X] Looked up the software bug tracker if reporting a build error?
- [X] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

#### Brief summary of issue:
I ran `brew update && brew upgrade` this morning and now `octave` is broken, yielding the following output:
```
• octave
dyld: Library not loaded: /usr/local/opt/readline/lib/libreadline.6.dylib
  Referenced from: /usr/local/bin/octave
  Reason: image not found
Abort trap: 6
```
`readline` was updated yesterday. Unfortunately, `libreadline.6.dylib` is linked by `octave` instead of `libreadline.dylib`, causing `octave` to break everytime `readline` is updated.

I understand that a short-term fix can be provided  by soft linking libreadline.6.dylib. It just seems to me that it is an error in the `octave` package to link to a specific version of a library. If this is intentional, then how can we prevent a version of a library from being removed by homebrew when it is upgraded if it is depended on by another formula?

#### The complete command that you issued:
`octave`

#### Gist with output of `brew doctor`:
https://gist.github.com/houtanb/230186ee2b089c74e9c9e39405248152

#### Link output by `brew gist-logs <formula>`:
```
• brew gist-logs octave
Error: No logs.
```

#### Output of `brew config`:
https://gist.github.com/houtanb/e02a66c7c38b8e63337205a242899bb7

**Comment by tomasdikk** [comment]
Having the exact same problem here unfortunately.

**Comment by dpo** [comment]
@schoeps Perhaps this can be addressed in #4327?

**Comment by ilovezfs** [comment]
I'm the one who broke it. It needs a revision bump due to the `readline` upgrade in core: https://github.com/Homebrew/homebrew-core/pull/4869

All science victims:

homebrew/science/gap                   
homebrew/science/gnudatalanguage       
homebrew/science/nest                  
homebrew/science/nexusformat           
homebrew/science/octave                
homebrew/science/pspp                  
homebrew/science/qcl                   
homebrew/science/r                     
homebrew/science/shogun                
homebrew/science/symphony

**Comment by ilovezfs** [comment]
Closing in favor of https://github.com/Homebrew/homebrew-science/issues/4367.

**Comment by ilovezfs** [comment]
Should now be fixed if you brew update and brew upgrade

**Comment by houtanb** [comment]
@ilovezfs thanks

**Comment by houtanb** [comment]
@ilovezfs actually, I'm a bit concerned, because octave was just upgraded to 4.2 rc2, which is not the current stable version. Is there any way to make this fix for Octave 4.0.3? I don't think that the default version of octave should be a release candidate

**Comment by dpo** [comment]
https://github.com/Homebrew/homebrew-science/pull/4327#issuecomment-250449428

**Comment by houtanb** [comment]
@dpo yes, but it hasn't been released yet as 4.2. What's more, for those of us who work on multiple platforms (I work on OS X and Debian), this creates an environment that is not consistent. It's just not correct to force upgrades before they are officially released, no matter if they will "probably" be released as is.

**Comment by ilovezfs** [comment]
@houtanb you're welcome to discuss this with @schoeps on #4327.
